projects/m1/027-the-collatz-conjecture/js/index.js

```diff
@@ -0,0 +1,43 @@
+// const prompt = require("prompt-sync")({ sigint: true}) ;
+
+function askInteger(){
+    const askedInteger = parseInt( prompt( 'Enter an integer number:', 'Example: 6' ) );
+
+    if ( ! isNaN( askedInteger ) ){
+        return askedInteger;
+    }
+
+    console.log( 'Please enter a valid integer number.' );
+    return askInteger();
+}
+
+function collatzConjecture( integerNumber ){
+
+    if ( integerNumber > 0 ){
+
+        while ( integerNumber !== 1 ) {
+            if ( integerNumber % 2 === 0 ){
+                // JavaScript hasn't a floor division operator like in Python (Double Slash Operator //) , then we apply Math.floor to the division result.
+                integerNumber = Math.floor( integerNumber / 2 );

```

Comment:
Watch out for the behavior of `math.floor`. 
Try `Math.floor(-100/3)` ;-)

I recommend `math.trunc` instead.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/trunc

Or symbolically for example:
`~~(-100/3)`

Reply:
Hi, I made the changes, thank you.